emodelrunner/protocols/create_protocols.py

```diff
@@ -0,0 +1,456 @@
+        # find threshold and holding currents
+        thres_i = None
+        hold_i = None
+        for key, resp in responses.items():
+            if "threshold_current" in key:
+                thres_i = resp
+            elif "holding_current" in key:
+                hold_i = resp
+
+        currents = {}
+        for protocol in self.protocols.protocols:
+            currents.update(
+                protocol.generate_current(
+                    threshold_current=thres_i, holding_current=hold_i, dt=dt
+                )
+            )
+
+        return currents
+
+    def get_thalamus_stim_currents(self, responses, mtype, dt):
+        """Returns the currents injected by thalamus protocols.
+
+        Args:
+            responses (dict): protocol responses
+            mtype (str): mtype to index the responses
+            dt (float): timestep of the generated currents (ms)
+
+        Returns:
+            dict: the generated currents in a dict.
+        """
+        thres_i_hyp = responses[f"{mtype}.bpo_threshold_current_hyp"]
+        thres_i_dep = responses[f"{mtype}.bpo_threshold_current_dep"]
+        holding_i_hyp = responses[f"{mtype}.bpo_holding_current_hyp"]
+        holding_i_dep = responses[f"{mtype}.bpo_holding_current_dep"]
+        currents = {}
+        for protocol in self.protocols.protocols:
+            currents.update(
+                protocol.generate_current(
+                    thres_i_hyp, thres_i_dep, holding_i_hyp, holding_i_dep, dt
+                )
+            )
+        return currents
+
+
+def create_protocols_object(
+    apical_point_isec,
+    prot_path,
+    package_type,
+    features_path="",
+    mtype="",
+    syn_locs=None,
+    stochkv_det=None,
+):
+    """Return a dict containing protocols.
+
+    Args:
+        apical_point_isec (int): section index of the apical point
+            Set to -1 no apical point is used in any extra recordings
+        prot_path (str): path to the protocols file
+        package_type (Enum): enum denoting the package type
+        features_path (str): path to the features file
+        mtype (str): morphology name to be used as prefix in output filenames
+        syn_locs (list): list of synapse locations
+        stochkv_det (bool): set if stochastic or deterministic
+
+    Returns:
+        ephys.protocols.SequenceProtocol: sequence protocol containing all the protocols
+    """
+    # pylint: disable=unbalanced-tuple-unpacking, too-many-locals
+    if package_type == PackageType.sscx:
+        protocols_dict = ProtocolParser().parse_sscx_protocols(
+            prot_path,
+            stochkv_det,
+            mtype,
+            apical_point_isec,
+            syn_locs,
+        )
+    elif package_type == PackageType.thalamus:
+        protocols_dict = ProtocolParser().parse_thalamus_protocols(
+            prot_path,
+            stochkv_det,
+            mtype,
+        )
+    else:
+        raise ValueError(f"unsupported package type: {package_type}")
```

Comment:
Could you just add `raises ValueError if package_type is not sscx neither thalamus` or something like that in the docstring?

Reply:
Aside from that it looks good, that would be the last change before the merge ;)

Reply:
Ok, added in 6214d5f